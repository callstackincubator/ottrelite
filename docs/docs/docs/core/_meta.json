[
  {
    "type": "file",
    "name": "introduction.mdx",
    "label": "Introduction"
  },
  {
    "type": "file",
    "name": "react-native-internals.mdx",
    "label": "React Native Internals"
  }
]
