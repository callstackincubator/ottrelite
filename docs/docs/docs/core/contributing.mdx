# Contributing

## Creating a backend

TODO: write this section.

## Development

To start developing, first clone the repository:

```bash
git clone https://github.com/callstackincubator/ReactNativeOttrelite
```

Then, install the dependencies & generate the glue code using Nitro Modules:

```bash
pnpm i
```

Any time you change the Nitro spec files (TS), run `pnpm run build` to regenerate the glue code.

### iOS - additional setup

For iOS, you need to install the CocoaPods dependencies. Run the following command in the `ios` directory of your project:

```bash
pnpm pod:install
```
