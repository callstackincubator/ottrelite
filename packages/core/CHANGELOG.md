# @ottrelite/core

## 0.1.1

### Patch Changes

- 727ccab: docs: updated READMEs

## 0.1.0

### Patch Changes

- 6f1463b: Initial release
