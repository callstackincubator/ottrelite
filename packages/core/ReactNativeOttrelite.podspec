require "json"

package = JSON.parse(File.read(File.join(__dir__, "package.json")))

Pod::Spec.new do |s|
  s.name         = "ReactNativeOttrelite"
  s.version      = package["version"]
  s.summary      = package["description"]
  s.homepage     = package["homepage"]
  s.license      = package["license"]
  s.authors      = package["author"]

  s.platforms    = { :ios => min_ios_version_supported, :visionos => 1.0 }
  s.source       = { :git => "https://github.com/callstackincubator/ottrelite.git" }
  
  s.pod_target_xcconfig = {
    # C++ compiler flags, mainly for folly.
    "GCC_PREPROCESSOR_DEFINITIONS" => "$(inherited) OTTRELITE_CORE_VERSION=\\\"#{package["version"]}\\\" FOLLY_NO_CONFIG FOLLY_CFG_NO_COROUTINES",
    # C++ standard flags
    "CLANG_CXX_LANGUAGE_STANDARD" => "c++17",
    "CLANG_CXX_LIBRARY" => "libc++"
  }

  # Ottrelite's sources
  s.source_files = [
    # Implementation (Swift)
    "ios/**/*.{swift}",
    # Autolinking/Registration (Objective-C++)
    "ios/**/*.{m,mm,hpp,h}",
    # Implementation (C++ objects)
    "cpp/**/*.{h,hpp,c,cpp}"
  ]

  # Expose headers
  s.public_header_files = [
    "cpp/**/*.{h,hpp}",
    "ios/**/*.{h,hpp}"
  ]

  load 'nitrogen/generated/ios/ReactNativeOttrelite+autolinking.rb'
  add_nitrogen_files(s)

  s.dependency 'React-jsi'
  s.dependency 'React-callinvoker'

  install_modules_dependencies(s)
end
