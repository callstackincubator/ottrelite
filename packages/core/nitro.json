{
  "$schema": "https://nitro.margelo.com/nitro.schema.json",
  "cxxNamespace": ["ottrelite"],
  "ios": {
    "iosModuleName": "ReactNativeOttrelite"
  },
  "android": {
    "androidNamespace": ["ottrelite"],
    "androidCxxLibName": "ReactNativeOttrelite"
  },
  "autolinking": {
    "ReactNativeOttrelite": {
      "cpp": "ReactNativeOttrelite"
    },
    "ReactNativeOttreliteExporterOTLP": {
      "cpp": "ReactNativeOttreliteExporterOTLP"
    }
  },
  "ignorePaths": ["**/node_modules"]
}
