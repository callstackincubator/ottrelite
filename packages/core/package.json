{
  "name": "@ottrelite/core",
  "version": "0.0.1",
  "description": "Extensible, unified tracing framework for React Native",
  "main": "lib/index",
  "module": "lib/index",
  "types": "lib/index.d.ts",
  "react-native": "src/index",
  "source": "src/index",
  "author": "Callstack <hello@callstack.com>",
  "files": [
    "src",
    "react-native.config.js",
    "lib",
    "nitrogen",
    "android/build.gradle",
    "android/gradle.properties",
    "android/CMakeLists.txt",
    "android/src",
    "cpp",
    "ios/**/*.h",
    "ios/**/*.m",
    "ios/**/*.mm",
    "ios/**/*.cpp",
    "ios/**/*.swift",
    "*.podspec",
    "README.md"
  ],
  "scripts": {
    "postinstall": "pnpm run build || exit 0;",
    "typecheck": "tsc --noEmit",
    "clean": "del-cli android/build node_modules/**/android/build lib",
    "lint": "eslint \"**/*.{js,ts,tsx}\" --fix",
    "lint-ci": "eslint \"**/*.{js,ts,tsx}\" -f @jamesacarr/github-actions",
    "typescript": "tsc",
    "build": "nitro-codegen --logLevel=\"debug\" && pnpm run typescript",
    "build:watch": "tsc --watch",
    "publish:npm": "pnpm publish --access public",
    "publish:yalc": "yalc publish"
  },
  "keywords": [
    "react-native",
    "tracing",
    "trace",
    "observability",
    "performance"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/callstackincubator/ottrelite.git"
  },
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/callstackincubator/ottrelite/issues"
  },
  "homepage": "https://github.com/callstackincubator/ottrelite#readme",
  "publishConfig": {
    "registry": "https://registry.npmjs.org/"
  },
  "devDependencies": {
    "@babel/runtime": "^7.25.0",
    "@opentelemetry/api": "^1.9.0",
    "del-cli": "^6.0.0",
    "eslint": "^8.57.1",
    "jest": "^30.0.4",
    "nitro-codegen": "^0.27.2",
    "react-native-nitro-modules": "0.29.6",
    "typescript": "^5.8.3",
    "react-native": "^0.81.0",
    "react": "19.1.0"
  },
  "peerDependencies": {
    "react": ">=17",
    "react-native": ">=0.76",
    "react-native-nitro-modules": ">=0.29"
  }
}
