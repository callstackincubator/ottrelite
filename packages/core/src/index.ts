import { Ottrelite } from './Ottrelite';

export type { OttreliteBackendInfo } from './types/OttreliteBackendInfo';
export type { OttreliteBackend } from './types/OttreliteBackend';

export * from './otel';
export * from './hooks';
export * from './types';
export * from './hoc';

export { Ottrelite } from './Ottrelite';
export default Ottrelite;
