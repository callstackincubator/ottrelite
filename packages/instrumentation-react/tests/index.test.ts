import { transform } from '@babel/core';

import {
  CORE_PACKAGE_NAME,
  TRACE_COMPONENT_LIFECYCLE_HOC_NAME,
  TRACE_COMPONENT_LIFECYCLE_HOOK_NAME,
} from '../src/constants';
import ottrelitePlugin from '../src/index';

function transformCode(code: string) {
  const result = transform(code, {
    presets: ['module:@react-native/babel-preset'],
    plugins: [ottrelitePlugin],
    filename: 'test.js',
    compact: false,
    retainLines: false,
  });

  return result?.code || '';
}

function normalizeCode(code: string) {
  return (
    code
      // replace UUID-based names with placeholders for consistent snapshots
      .replaceAll(/(\S*)_[a-f0-9]{32}/g, '$1_UUID_PLACEHOLDER')
      // normalize whitespace
      .trim()
  );
}

describe('Ottrelite Babel Plugin', () => {
  describe('Transform tests', () => {
    test('should transform function declaration with implicit tracer name', () => {
      const code = `
        function MyComponent() {
          'use trace';
          const x = 1;
          return <div>{x}</div>;
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform a complex component with implicit tracer name', () => {
      const code = `
        import { useSubheading } from '@hooks/useSubheading';
        import { useEffect, useState } from 'react';
        import { View } from 'react-native';
        import { Text } from 'react-native-paper';

        export type MyComponentProps = {
          heading: string;
        };

        export function MyComponent({ heading }: MyComponentProps) {
          'use trace';

          const [res, setRes] = useState<{ count: number } | null>(null);
          const { subheading } = useSubheading();

          useEffect(() => {
            fetch('http://api.test.com/data')
              .then((response) => response.json())
              .then((data) => setRes(data))
              .catch((error) => console.error(error));
          }, []);

          return (
            <View>
              <Text variant="titleLarge">{heading}</Text>

              <Text variant="titleSmall">{subheading}</Text>

              <Text>{res?.count ?? 'Loading...'}</Text>
            </View>
          );
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform function with explicit tracer name', () => {
      const code = `
        function MyComponent() {
          'use trace CustomName';
          return <div>Hello</div>;
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform arrow function assigned to variable with implicit tracer name', () => {
      const code = `
        const MyComponent = () => {
          'use trace';
          return <div>Hello</div>;
        };
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform function expression assigned to variable with implicit tracer name', () => {
      const code = `
        const MyComponent = function() {
          'use trace';
          return <div>Hello</div>;
        };
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform class methods with implicit tracer name', () => {
      const code = `
        class MyClass {
          render() {
            'use trace';
            return 'result';
          }
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform class component with unary explicit dev API', () => {
      const code = `
        class MyClass {
          render() {
            'use trace dev';
            return 'result';
          }
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform class component with unary explicit otel API', () => {
      const code = `
        class MyClass {
          render() {
            'use trace otel';
            return 'result';
          }
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform class component with unary custom component name', () => {
      const code = `
        class MyClass {
          render() {
            'use trace CustomTestName';
            return 'result';
          }
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform class components with two parameters: API and component name, for both APIs specified', () => {
      const code = `
        class MyClass {
          render() {
            'use trace dev CustomTestName1';
            return 'result';
          }
        }
        class MyClass2 {
          render() {
            'use trace otel CustomTestName2';
            return 'result';
          }
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform function component with unary explicit dev API', () => {
      const code = `
        function MyComponent() {
          'use trace dev';
          return 'result';
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform function component with unary explicit otel API', () => {
      const code = `
        function MyComponent() {
          'use trace otel';
          return 'result';
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform function component with unary custom component name', () => {
      const code = `
        function MyComponent() {
          'use trace CustomTestName';
          return 'result';
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform function components with two parameters: API and component name', () => {
      const code = `
        function MyComponent() {
          'use trace dev CustomTestName';
          return 'result';
        }
        function MyComponent2() {
          'use trace otel CustomTestName';
          return 'result';
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should transform multiple functions in same file with implicit tracer names', () => {
      const code = `
        function ComponentA() {
          'use trace';
          return <div>A</div>;
        }
        
        function ComponentB() {
          'use trace';
          return <div>B</div>;
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should handle function with complex body with implicit tracer name', () => {
      const code = `
        function MyComponent(props) {
          'use trace';
          const x = 1;
          const y = 2;
          console.log(x + y);
          if (props.show) {
            return <div>{x + y}</div>;
          }
          return null;
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should handle function with complex body and destructured props with implicit tracer name', () => {
      const code = `
        function MyComponent({ show }) {
          'use trace';
          const x = 1;
          const y = 2;
          console.log(x + y);
          if (show) {
            return <div>{x + y}</div>;
          }
          return null;
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should not transform function without use trace directive', () => {
      const code = `
        function MyComponent() {
          'use strict';
          return <div>Hello</div>;
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    test('should not transform when no directive is present', () => {
      const code = `
        function MyComponent() {
          const x = 1;
          return <div>{x}</div>;
        }
      `;

      const result = transformCode(code);
      expect(normalizeCode(result)).toMatchSnapshot();
    });

    describe('Error handling', () => {
      test('should throw error for anonymous default export', () => {
        const code = `
          export default function() {
            'use trace';
            return <div>Hello</div>;
          }
        `;

        expect(() => transformCode(code)).toThrow(
          /is an anonymous default export, which is not supported/
        );
      });
    });

    describe('Import behavior', () => {
      test(`should add ${TRACE_COMPONENT_LIFECYCLE_HOOK_NAME} import when use trace directive is found in an FC`, () => {
        const code = `
          function MyComponent() {
            'use trace';
            return <div>Hello</div>;
          }
        `;

        const result = transformCode(code);
        expect(result).toContain(CORE_PACKAGE_NAME);
        expect(result).toContain(TRACE_COMPONENT_LIFECYCLE_HOOK_NAME);
      });

      test(`should not add duplicate import if ${TRACE_COMPONENT_LIFECYCLE_HOOK_NAME} is already imported an FC`, () => {
        const code = `
          import { ${TRACE_COMPONENT_LIFECYCLE_HOOK_NAME} } from '${CORE_PACKAGE_NAME}';
          
          function MyComponent() {
            'use trace';
            return <div>Hello</div>;
          }
        `;

        const result = transformCode(code);
        const importMatches = result.match(/@ottrelite\/core/g);
        expect(importMatches).toHaveLength(1);
      });

      test('should not add import when no use trace directive is found an FC', () => {
        const code = `
          function MyComponent() {
            return <div>Hello</div>;
          }
        `;

        const result = transformCode(code);
        expect(result).not.toContain(CORE_PACKAGE_NAME);
      });

      test(`should add ${TRACE_COMPONENT_LIFECYCLE_HOOK_NAME} import when use trace directive is found in a class component`, () => {
        const code = `
          class MyComponent {
            render() {
              'use trace';
              return <div>Hello</div>;
            }
          }
        `;

        const result = transformCode(code);
        expect(result).toContain(CORE_PACKAGE_NAME);
        expect(result).toContain(TRACE_COMPONENT_LIFECYCLE_HOC_NAME);
      });

      test(`should not add duplicate import if ${TRACE_COMPONENT_LIFECYCLE_HOOK_NAME} is already imported in a class component`, () => {
        const code = `
          import { ${TRACE_COMPONENT_LIFECYCLE_HOOK_NAME} } from '${CORE_PACKAGE_NAME}';
          
          class MyComponent {
            render() {
              'use trace';
              return <div>Hello</div>;
            }
          }
        `;

        const result = transformCode(code);
        const importMatches = result.match(/@ottrelite\/core/g);
        expect(importMatches).toHaveLength(1);
      });

      test('should not add import when no use trace directive is found in a class component', () => {
        const code = `
          class MyComponent {
            render() {
              return <div>Hello</div>;
            }
          }
        `;

        const result = transformCode(code);
        expect(result).not.toContain(CORE_PACKAGE_NAME);
      });
    });
  });
});
