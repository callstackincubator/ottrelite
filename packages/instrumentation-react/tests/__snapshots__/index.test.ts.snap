// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Ottrelite Babel Plugin Transform tests should handle function with complex body and destructured props with implicit tracer name 1`] = `
"var _core = require("@ottrelite/core");
var _jsxRuntime = require("react/jsx-runtime");
function MyComponent(_ref) {
  var show = _ref.show;
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("MyComponent", undefined, "dev");
  function OttreliteWrappedMyComponent(_ref) {
    var x = 1;
    var y = 2;
    console.log(x + y);
    if (show) {
      return (0, _jsxRuntime.jsx)("div", {
        children: x + y
      });
    }
    return null;
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedMyComponent();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
}"
`;

exports[`Ottrelite Babel Plugin Transform tests should handle function with complex body with implicit tracer name 1`] = `
"var _core = require("@ottrelite/core");
var _jsxRuntime = require("react/jsx-runtime");
function MyComponent(props) {
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("MyComponent", undefined, "dev");
  function OttreliteWrappedMyComponent(props) {
    var x = 1;
    var y = 2;
    console.log(x + y);
    if (props.show) {
      return (0, _jsxRuntime.jsx)("div", {
        children: x + y
      });
    }
    return null;
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedMyComponent(props);
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
}"
`;

exports[`Ottrelite Babel Plugin Transform tests should not transform function without use trace directive 1`] = `
"var _jsxRuntime = require("react/jsx-runtime");
function MyComponent() {
  'use strict';

  return (0, _jsxRuntime.jsx)("div", {
    children: "Hello"
  });
}"
`;

exports[`Ottrelite Babel Plugin Transform tests should not transform when no directive is present 1`] = `
"var _jsxRuntime = require("react/jsx-runtime");
function MyComponent() {
  var x = 1;
  return (0, _jsxRuntime.jsx)("div", {
    children: x
  });
}"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform a complex component with implicit tracer name 1`] = `
"var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MyComponent = MyComponent;
var _core = require("@ottrelite/core");
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _useSubheading2 = require("@hooks/useSubheading");
var _react = require("react");
var _reactNative = require("react-native");
var _reactNativePaper = require("react-native-paper");
var _jsxRuntime = require("react/jsx-runtime");
function MyComponent(_ref) {
  var heading = _ref.heading;
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("MyComponent", undefined, "dev");
  function OttreliteWrappedMyComponent(_ref) {
    var _res$count;
    var _ref2 = _react.useState < {
        count: number
      } | null > null,
      _ref3 = (0, _slicedToArray2.default)(_ref2, 2),
      res = _ref3[0],
      setRes = _ref3[1];
    var _useSubheading = (0, _useSubheading2.useSubheading)(),
      subheading = _useSubheading.subheading;
    (0, _react.useEffect)(function () {
      fetch('http://api.test.com/data').then(function (response) {
        return response.json();
      }).then(function (data) {
        return setRes(data);
      }).catch(function (error) {
        return console.error(error);
      });
    }, []);
    return (0, _jsxRuntime.jsxs)(_reactNative.View, {
      children: [(0, _jsxRuntime.jsx)(_reactNativePaper.Text, {
        variant: "titleLarge",
        children: heading
      }), (0, _jsxRuntime.jsx)(_reactNativePaper.Text, {
        variant: "titleSmall",
        children: subheading
      }), (0, _jsxRuntime.jsx)(_reactNativePaper.Text, {
        children: (_res$count = res == null ? void 0 : res.count) != null ? _res$count : 'Loading...'
      })]
    });
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedMyComponent();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
}"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform arrow function assigned to variable with implicit tracer name 1`] = `
"var _core = require("@ottrelite/core");
var _jsxRuntime = require("react/jsx-runtime");
var MyComponent = function MyComponent() {
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("MyComponent", undefined, "dev");
  function OttreliteWrappedMyComponent() {
    return (0, _jsxRuntime.jsx)("div", {
      children: "Hello"
    });
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedMyComponent();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
};"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform class component with unary custom component name 1`] = `
"var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _core = require("@ottrelite/core");
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var MyClass = (0, _core.withComponentRenderTracing)("CustomTestName", "dev", function () {
  function MyClass() {
    (0, _classCallCheck2.default)(this, MyClass);
  }
  return (0, _createClass2.default)(MyClass, [{
    key: "render",
    value: function render() {
      return 'result';
    }
  }]);
}());"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform class component with unary explicit dev API 1`] = `
"var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _core = require("@ottrelite/core");
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var MyClass = (0, _core.withComponentRenderTracing)("MyClass", "dev", function () {
  function MyClass() {
    (0, _classCallCheck2.default)(this, MyClass);
  }
  return (0, _createClass2.default)(MyClass, [{
    key: "render",
    value: function render() {
      return 'result';
    }
  }]);
}());"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform class component with unary explicit otel API 1`] = `
"var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _core = require("@ottrelite/core");
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var MyClass = (0, _core.withComponentRenderTracing)("MyClass", "otel", function () {
  function MyClass() {
    (0, _classCallCheck2.default)(this, MyClass);
  }
  return (0, _createClass2.default)(MyClass, [{
    key: "render",
    value: function render() {
      return 'result';
    }
  }]);
}());"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform class components with two parameters: API and component name, for both APIs specified 1`] = `
"var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _core = require("@ottrelite/core");
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var MyClass = (0, _core.withComponentRenderTracing)("CustomTestName1", "dev", function () {
  function MyClass() {
    (0, _classCallCheck2.default)(this, MyClass);
  }
  return (0, _createClass2.default)(MyClass, [{
    key: "render",
    value: function render() {
      return 'result';
    }
  }]);
}());
var MyClass2 = (0, _core.withComponentRenderTracing)("CustomTestName2", "otel", function () {
  function MyClass2() {
    (0, _classCallCheck2.default)(this, MyClass2);
  }
  return (0, _createClass2.default)(MyClass2, [{
    key: "render",
    value: function render() {
      return 'result';
    }
  }]);
}());"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform class methods with implicit tracer name 1`] = `
"var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _core = require("@ottrelite/core");
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var MyClass = (0, _core.withComponentRenderTracing)("MyClass", "dev", function () {
  function MyClass() {
    (0, _classCallCheck2.default)(this, MyClass);
  }
  return (0, _createClass2.default)(MyClass, [{
    key: "render",
    value: function render() {
      return 'result';
    }
  }]);
}());"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform function component with unary custom component name 1`] = `
"var _core = require("@ottrelite/core");
function MyComponent() {
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("CustomTestName", undefined, "dev");
  function OttreliteWrappedCustomTestName() {
    return 'result';
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedCustomTestName();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
}"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform function component with unary explicit dev API 1`] = `
"var _core = require("@ottrelite/core");
function MyComponent() {
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("MyComponent", undefined, "dev");
  function OttreliteWrappedMyComponent() {
    return 'result';
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedMyComponent();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
}"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform function component with unary explicit otel API 1`] = `
"var _core = require("@ottrelite/core");
function MyComponent() {
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("MyComponent", undefined, "otel");
  function OttreliteWrappedMyComponent() {
    return 'result';
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedMyComponent();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
}"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform function components with two parameters: API and component name 1`] = `
"var _core = require("@ottrelite/core");
function MyComponent() {
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("CustomTestName", undefined, "dev");
  function OttreliteWrappedCustomTestName() {
    return 'result';
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedCustomTestName();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
}
function MyComponent2() {
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("CustomTestName", undefined, "otel");
  function OttreliteWrappedCustomTestName() {
    return 'result';
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedCustomTestName();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
}"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform function declaration with implicit tracer name 1`] = `
"var _core = require("@ottrelite/core");
var _jsxRuntime = require("react/jsx-runtime");
function MyComponent() {
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("MyComponent", undefined, "dev");
  function OttreliteWrappedMyComponent() {
    var x = 1;
    return (0, _jsxRuntime.jsx)("div", {
      children: x
    });
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedMyComponent();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
}"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform function expression assigned to variable with implicit tracer name 1`] = `
"var _core = require("@ottrelite/core");
var _jsxRuntime = require("react/jsx-runtime");
var MyComponent = function MyComponent() {
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("MyComponent", undefined, "dev");
  function OttreliteWrappedMyComponent() {
    return (0, _jsxRuntime.jsx)("div", {
      children: "Hello"
    });
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedMyComponent();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
};"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform function with explicit tracer name 1`] = `
"var _core = require("@ottrelite/core");
var _jsxRuntime = require("react/jsx-runtime");
function MyComponent() {
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("CustomName", undefined, "dev");
  function OttreliteWrappedCustomName() {
    return (0, _jsxRuntime.jsx)("div", {
      children: "Hello"
    });
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedCustomName();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
}"
`;

exports[`Ottrelite Babel Plugin Transform tests should transform multiple functions in same file with implicit tracer names 1`] = `
"var _core = require("@ottrelite/core");
var _jsxRuntime = require("react/jsx-runtime");
function ComponentA() {
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("ComponentA", undefined, "dev");
  function OttreliteWrappedComponentA() {
    return (0, _jsxRuntime.jsx)("div", {
      children: "A"
    });
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedComponentA();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
}
function ComponentB() {
  var __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER = (0, _core.useComponentRenderTracing)("ComponentB", undefined, "dev");
  function OttreliteWrappedComponentB() {
    return (0, _jsxRuntime.jsx)("div", {
      children: "B"
    });
  }
  var __ottreliteCompOrigResult_UUID_PLACEHOLDER = OttreliteWrappedComponentB();
  __ottreliteUseComponentRenderTracingResult_UUID_PLACEHOLDER.markJSRenderEnd();
  return __ottreliteCompOrigResult_UUID_PLACEHOLDER;
}"
`;
